spec/services/authentication/authenticator_spec.rb

```diff
@@ -221,6 +221,8 @@
     # Freeze time since `facebook_created_at` will be based on server time
     before do
       Timecop.freeze
+      allow_any_instance_of(Authentication::Providers::Facebook)
+          .to receive(:image_url).and_return("https://dummyimage.com/400x400.jpg")
```

Comment:
So I force pushed this even though Rubocop didn't like it. I'm not sure how to avoid using `allow_any_instance_of` here though. Would I do something like:
```ruby
let!(:facebook_provider) { instance_double(Authentication::Providers::Facebook) }
```

at the top? I think this doesn't work though because each authentication request has a new instance. 🤔

Reply:
What if instead of stubbing `image_url` you stub `HTTParty` and let it return what you need to make the test pass?

Reply:
Yeah, that's why `instance_double` doesn't work: each auth request gets a new instance.

I chose to use `allow_any_instance_of` because I was not interested in mocking the HTTP result. If you do go that route: consider using VCR to get an accurate representation of what Facebook is returning.